document/src/vespa/document/base/idstring.h

```diff
@@ -13,108 +12,60 @@ namespace document {
  * \class document::IdString
  * \ingroup base
  *
- * \brief Superclass for all document identifier schemes.
+ * \brief New scheme for documents with no forced distribution.
+ *
+ * By using this scheme, documents will be evenly distributed within VDS,
+ * as the location of a doc identifier is a hash of the entire URI.
+ * This scheme also contains the DocumentType.
  */
-class IdString : public vespalib::Cloneable {
+class IdString {
 public:
-    typedef std::unique_ptr<IdString> UP;
-    typedef vespalib::CloneablePtr<IdString> CP;
     typedef uint64_t LocationType;
-    enum Type { ID=0, NULLID };
-    static const vespalib::string & getTypeName(Type t);
-
-    /** @throws document::IdParseException If parsing of id scheme failed. */
-    static IdString::UP createIdString(vespalib::stringref id) { return createIdString(id.data(), id.size()); }
-    static IdString::UP createIdString(const char *id, size_t sz);
     static LocationType makeLocation(vespalib::stringref s);
 
-    ~IdString();
-    IdString* clone() const override = 0;
+    explicit IdString(vespalib::stringref ns);
+    IdString();
 
-    virtual Type        getType() const = 0;
     vespalib::stringref getNamespace() const { return getComponent(0); }
-    virtual vespalib::stringref getNamespaceSpecific() const = 0;
-    virtual LocationType getLocation() const = 0;
-    virtual std::pair<int16_t, int64_t> getGidBitsOverride() const { return std::pair<int16_t, int64_t>(0, 0); }
-    virtual bool hasDocType() const { return false; }
-    virtual vespalib::stringref getDocType() const { return ""; }
-    virtual bool hasNumber() const { return false; }
-    virtual uint64_t getNumber() const { return 0; }
-    virtual bool hasGroup() const { return false; }
-    virtual vespalib::stringref getGroup() const { return ""; }
+    bool hasDocType() const { return size(1 != 0); }
```

Comment:
Was this intended to be `return size(1) != 0;`? Funnily enough, due to implicit type conversions this ends up being the same expression in practice

Reply:
Good catch. this was a nasty little bugger. That was what I thought I wrote. At the test started working again :)